kratos/expression/integration_point_expression_io.h

```diff
@@ -0,0 +1,158 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+#pragma once
+
+// System incldues
+#include <variant>
+
+// Project includes
+#include "containers/array_1d.h"
+#include "expression/container_expression.h"
+#include "expression/traits.h"
+#include "containers/variable.h"
+#include "expression_io.h"
+#include "includes/define.h"
+#include "includes/model_part.h"
+
+namespace Kratos {
+
+
+class KRATOS_API(KRATOS_CORE) IntegrationPointExpressionIO
+{
+public:
+    ///@name Type definitions
+    ///@{
+
+    using VariableType = std::variant<
+                                const Variable<int>*,
+                                const Variable<double>*,
+                                const Variable<array_1d<double, 3>>*,
+                                const Variable<array_1d<double, 4>>*,
+                                const Variable<array_1d<double, 6>>*,
+                                const Variable<array_1d<double, 9>>*,
+                                const Variable<Vector>*,
+                                const Variable<Matrix>*>;
+
+
+    ///}
+    ///@name Public classes
+    ///@{
+
+    class KRATOS_API(KRATOS_CORE) IntegrationPointExpressionInput : public ExpressionInput
+    {
+    public:
+        ///@name Type definitions
+        ///@{
+
+        KRATOS_CLASS_POINTER_DEFINITION(IntegrationPointExpressionInput);
+
+        ///@}
+        ///@name Life cycle
+        ///@{
+
+        IntegrationPointExpressionInput(
+            ModelPart& rModelPart,
+            const VariableType& rVariable,
+            const ContainerType& rContainerType,
+            const MeshType& rMeshType = MeshType::Local);
+
+        ~IntegrationPointExpressionInput() override = default;
+
+        ///@}
+        ///@name Public operations
+        ///@{
+
+        Expression::Pointer Execute() const override;
+
+        ///@}
+
+    private:
+        ///@name Private member variables
+        ///@{
+
+        ModelPart* mpModelPart;
```

Comment:
shouldn't this be immutable?

Reply:
the `CalculateOnIntegrationPoints` only has non-const versions, therefore this needs to be mutable.